vault/login_mfa.go

```diff
@@ -2336,7 +2341,7 @@ func (c *Core) validatePingID(ctx context.Context, mConfig *mfa.Config, username
 }
 
 func (c *Core) validateTOTP(ctx context.Context, mfaFactors *MFAFactor, entityMethodSecret *mfa.Secret, configID, entityID string, usedCodes *cache.Cache, maximumValidationAttempts uint32) error {
-	if mfaFactors.passcode == "" {
+	if mfaFactors == nil || mfaFactors.passcode == "" {
```

Comment:
Why check for nil here, instead of at the caller?  Doing at the caller would allow us to see at the place where the nil,nil is returned that it's being handled.